### ISSUE TITLE
HTTP SD does not remove disappeared target groups

### ISSUE THREAD

**Issue by tksm** [prometheus/prometheus#9019]
**What did you do?**
I removed some target groups from my [HTTP SD](https://prometheus.io/docs/prometheus/latest/http_sd/) endpoint.

For example, the endpoint returns three target groups (`mylabel` :`v1`, `v2`, `v3`) at the first scraping.

```json
[
  {"labels": {"mylabel": "v1"}, "targets": ["127.0.0.1:9090"]},
  {"labels": {"mylabel": "v2"}, "targets": ["127.0.0.1:9090"]},
  {"labels": {"mylabel": "v3"}, "targets": ["127.0.0.1:9090"]}
]
```

Then the endpoint returns only one target group (`v1`) at the second scraping.

```json
[
  {"labels": {"mylabel": "v1"}, "targets": ["127.0.0.1:9090"]}
]
```

**What did you expect to see?**

HTTP SD removes disappeared target groups from scrape targets. In the case of the example, `v2`, `v3` should be deleted.

**What did you see instead? Under which circumstances?**

I still see disappeared target groups (`v2`, `v3` in the example) in scrape targets.

HTTP SD may need to send empty updates for deletion as [File SD does](https://github.com/prometheus/prometheus/blob/v2.28.0/discovery/file/file.go#L350-L365).

**Environment**

* Prometheus version: `prom/prometheus:v2.28.0`
* Prometheus configuration file:
```yaml
scrape_configs:
  - job_name: http_sd_job
    http_sd_configs:
    - url: http://localhost:8080/targets.json
      refresh_interval: 5s
```

**Comment by roidelapluie** [comment]
Fixed by #9020, released in 2.28.1

**Comment by tksm** [comment]
@roidelapluie @servak Thanks for the quick fix and release!
I confirmed that this issue is fixed in Prometheus v2.28.1.
